NTumbleBit/Proof/PoupardStern/Implementation/PoupardStern.cs

```diff
@@ -160,17 +160,23 @@ public static bool VerifyPoupardStern(this RsaKeyParameters pubKey, PoupardStern
             rPrime = y.Subtract(Modulus.Multiply(w));
 
             // Verifying x values
+            Dictionary<int, bool> numbers = new Dictionary<int, bool>();
```

Comment:
Use array `var numbers = new bool[BigK] ;`

Reply:
Changed to bool array with a comment that all elements in the array will get initialised to false. The logic within the loop has now changed to set array elements to `true`.